sunpy/net/helioviewer.py

```diff
@@ -109,14 +110,8 @@ def get_closest_image(self, date, observatory=None, instrument=None,
         >>> print(metadata['date'])  # doctest: +REMOTE_DATA
         2012-01-01T00:00:07.000
         """
-        if source_id is None:
-            try:
-                key = (observatory, instrument, detector, measurement)
-                source_id = self.data_sources[key]
-            except KeyError:
-                raise KeyError("The values used for observatory, instrument, detector, measurement "
-                               "do not correspond to a source_id. Please check the list using "
-                               "HelioviewerClient.data_sources.")
+        key = (observatory, instrument, detector, measurement)
+        source_id = self._get_source_id(source_id, key)
```

Comment:
Can we not pass the above line into this function?

Reply:
We can do this but then I guess we should pass key as a keyword argument

Reply:
Why would we need to pass it in as a keyword?

Reply:
I mean it will be better I guess to pass it as a keyword argument just for clear understanding

Reply:
I don't think so but I could be persuaded otherwise.

Reply:
Ok, no problem. We won't keep it as a keyword argument